src/protobuf/cosmos/base/v1beta1/coin.ts

```diff
@@ -135,7 +135,7 @@ export const DecCoin = {
           message.denom = reader.string();
           break;
         case 2:
-          message.amount = reader.string();
+          message.amount = (parseInt(reader.string()) / (10 ** 18)).toPrecision(18);
```

Comment:
The actual meat-and-potatoes of this fix. Turns `2000000000000000000` into `2.000000000000000000`.